src/views/virtualmachines/details/components/details/utils/operatingSystemHelper.ts

```diff
@@ -0,0 +1,31 @@
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+
+import { NAME_OS_TEMPLATE_ANNOTATION, OS_TEMPLATE_LABEL } from '../../../../utils/constants';
+
+type StringHashMap = {
+  [key: string]: string;
+};
+
+const getPrefixedKey = (obj: StringHashMap, keyPrefix: string) =>
+  obj ? Object.keys(obj).find((key) => key.startsWith(keyPrefix)) : null;
+
+const getSuffixValue = (key: string) => {
+  const index = key ? key.lastIndexOf('/') : -1;
+  return index > 0 ? key.substring(index + 1) : null;
+};
+
+export const findKeySuffixValue = (obj: StringHashMap, keyPrefix: string) =>
+  getSuffixValue(getPrefixedKey(obj, keyPrefix));
+
+export const getValueByPrefix = (obj: StringHashMap, keyPrefix: string): string => {
+  const objectKey = Object.keys(obj || {}).find((key) => key.startsWith(keyPrefix));
+  return objectKey ? obj[objectKey] : null;
+};
+
+export const getOperatingSystem = (obj: K8sResourceCommon): string =>
```

Comment:
If you know this is an operating system, why not use `os` instead of `obj`?

Reply:
the obj will be something general like vm/vmi/template I probably should move this the src/utils later on in a follow-up PR :)